plugin/rewrite/name.go

```diff
@@ -115,76 +116,141 @@ func (rule *regexNameRule) Rewrite(ctx context.Context, state request.Request) R
 
 // newNameRule creates a name matching rule based on exact, partial, or regex match
 func newNameRule(nextAction string, args ...string) (Rule, error) {
+	var matchType, rewriteQuestionFrom, rewriteQuestionTo string
+	var rewriteAnswerField, rewriteAnswerFrom, rewriteAnswerTo string
 	if len(args) < 2 {
 		return nil, fmt.Errorf("too few arguments for a name rule")
 	}
-	if len(args) == 3 {
-		switch strings.ToLower(args[0]) {
+	if len(args) == 2 {
+		matchType = "exact"
+		rewriteQuestionFrom = plugin.Name(args[0]).Normalize()
+		rewriteQuestionTo = plugin.Name(args[1]).Normalize()
+	}
+	if len(args) >= 3 {
+		matchType = strings.ToLower(args[0])
+		rewriteQuestionFrom = plugin.Name(args[1]).Normalize()
+		rewriteQuestionTo = plugin.Name(args[2]).Normalize()
+	}
+	if matchType == RegexMatch {
+		rewriteQuestionFrom = args[1]
+		rewriteQuestionTo = args[2]
+	}
+	if matchType == ExactMatch || matchType == SuffixMatch {
+		if !hasClosingDot(rewriteQuestionFrom) {
+			rewriteQuestionFrom = rewriteQuestionFrom + "."
+		}
+		if !hasClosingDot(rewriteQuestionTo) {
+			rewriteQuestionTo = rewriteQuestionTo + "."
+		}
+	}
+
+	if len(args) > 3 && len(args) != 7 {
+		return nil, fmt.Errorf("response rewrites must consist only of a name rule with 3 arguments and an answer rule with 3 arguments")
+	}
+
+	if len(args) < 7 {
+		switch matchType {
 		case ExactMatch:
-			return &nameRule{nextAction, plugin.Name(args[1]).Normalize(), plugin.Name(args[2]).Normalize()}, nil
+			rewriteAnswerFromPattern, err := isValidRegexPattern(rewriteQuestionTo, rewriteQuestionFrom)
+			if err != nil {
+				return nil, err
+			}
+			return &exactNameRule{
+				nextAction,
+				rewriteQuestionFrom,
+				rewriteQuestionTo,
+				ResponseRule{
```

Comment:
I think if we check at client write time, that we can handle all cases (including regex), and we don't need to set up the response re-write rule at setup time (which only works for exact match).

Reply:
> I think if we check at client write time

@chrisohaver , what impact on the performance do you foresee?

Reply:
I dont foresee a significant impact on performance.